docroot/modules/custom/sitenow_signage/sitenow_signage.module

```diff
@@ -34,6 +34,83 @@ function sitenow_signage_preprocess_field(&$variables) {
       }
       break;
 
+    case 'field_astra_schedule_room':
+      // Cache this for 1 day.
+      $variables['#cache']['tags'][] = 'time:daily';
+      // Get additional info necessary to overwrite the output of this field.
+      $paragraph = $variables['element']['#object'];
+      $headline = $paragraph->get('field_headline')->value;
+      $today = date('Y-m-d');
+      $had_failure = FALSE;
+      $rows = [];
+      $table_caption = 'Room Schedule';
+
+      if (!empty($variables['items'][0]['content']['#markup'])) {
+        $value = $variables['items'][0]['content']['#markup'];
+        // Split field value into building code and room id.
+        [$building_code, $room_id] = explode('-', $value);
+
+        // Load and fetch data from the MAUI API.
+        $api_client = \Drupal::service('uiowa_maui.api');
+        $data = $api_client->getRoomSchedule($today, $today, $building_code, $room_id);
+
+        if ($data === FALSE) {
+          $had_failure = TRUE;
+        }
+        elseif (!empty($data)) {
+          foreach ($data as $event) {
+            $formatted_time = trim($event->startTime) . ' – ' . trim($event->endTime);
+            $rows[] = [
+              'time' => $formatted_time,
+              'event' => trim($event->title),
+            ];
+          }
+        }
+      }
+
+      $empty_message = $had_failure
+        ? t('There was an error retrieving room schedule information. Please try again later or contact the ITS Help Desk if the problem persists.')
+        : t('No scheduled events for this room.');
+
+      $header = [
+        'time' => t('Time'),
+        'event' => t('Event'),
+      ];
+      $markup['container'] = [
+        '#type' => 'container',
+        '#attributes' => [
+          'class' => [
+            'signage__table',
+          ],
+        ],
+      ];
+      // Headline overrides everything if set.
+      if (!empty($headline)) {
+        $table_caption = $headline;
+      }
+      // If no headline and no failure, use room-specific caption.
+      elseif (!$had_failure && !empty($room_id)) {
+        $table_caption = 'Schedule for room ' . $room_id;
+      }
+
+      $markup['container']['table'] = [
+        '#theme' => 'table',
+        '#caption' => $table_caption,
+        '#header' => $header,
+        '#rows' => $rows,
+        '#empty' => $empty_message,
+        '#attributes' => [
+          'class' => [
+            'data',
```

Comment:
I missed this on in the BEM refactor.   Should this change to something we are modifying about the table like `signage__table--element-being-modified`?

Reply:
I removed the wrapper classes and just set the `signage__table` as the single class and it seems to be working.